lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java

```diff
@@ -3248,6 +3259,227 @@ void should_ReturnEmptyList_When_QueryingForNonexistentSystemInstanceLike() {
+                .customIntField(TaskCustomIntField.CUSTOM_INT_1, 1)
+                .customIntField(TaskCustomIntField.CUSTOM_INT_2, 2)
+                .customIntField(TaskCustomIntField.CUSTOM_INT_3, 3)
+                .customIntField(TaskCustomIntField.CUSTOM_INT_4, 4)
+                .customIntField(TaskCustomIntField.CUSTOM_INT_5, 5)
+                .customIntField(TaskCustomIntField.CUSTOM_INT_6, 6)
+                .customIntField(TaskCustomIntField.CUSTOM_INT_7, 7)
+                .customIntField(TaskCustomIntField.CUSTOM_INT_8, 8)
+                .buildAndStoreAsSummary(taskService);
+        taskSummary2 =
+            taskInWorkbasket(wb)
+                .customIntField(TaskCustomIntField.CUSTOM_INT_1, -1)
+                .customIntField(TaskCustomIntField.CUSTOM_INT_2, -2)
+                .customIntField(TaskCustomIntField.CUSTOM_INT_3, -3)
+                .customIntField(TaskCustomIntField.CUSTOM_INT_4, -4)
+                .customIntField(TaskCustomIntField.CUSTOM_INT_5, -5)
+                .customIntField(TaskCustomIntField.CUSTOM_INT_6, -6)
+                .customIntField(TaskCustomIntField.CUSTOM_INT_7, -7)
+                .customIntField(TaskCustomIntField.CUSTOM_INT_8, -8)
+                .buildAndStoreAsSummary(taskService);
+        taskSummary3 = taskInWorkbasket(wb).buildAndStoreAsSummary(taskService);
+      }
+
+      @WithAccessId(user = "user-1-1")
+      @TestFactory
+      Stream<DynamicTest> should_ThrowException_When_QueryingForCustomIntWithinInvalidInterval() {
+        List<Pair<TaskCustomIntField, IntInterval[]>> testCases =
+            List.of(
+                Pair.of(TaskCustomIntField.CUSTOM_INT_1, new IntInterval[] {new IntInterval(4, 1)}),
+                // Only first interval invalid
+                Pair.of(
+                    TaskCustomIntField.CUSTOM_INT_2,
+                    new IntInterval[] {new IntInterval(null, null), new IntInterval(0, null)}),
+                // Only second interval invalid
+                Pair.of(
+                    TaskCustomIntField.CUSTOM_INT_3,
+                    new IntInterval[] {new IntInterval(-1, 5), new IntInterval(null, null)}),
+                // Both intervals invalid
+                Pair.of(
+                    TaskCustomIntField.CUSTOM_INT_4,
+                    new IntInterval[] {new IntInterval(0, -5), new IntInterval(-2, -10)}),
+                Pair.of(
+                    TaskCustomIntField.CUSTOM_INT_5,
+                    new IntInterval[] {new IntInterval(null, null)}),
+                Pair.of(
+                    TaskCustomIntField.CUSTOM_INT_6, new IntInterval[] {new IntInterval(0, -5)}),
+                Pair.of(
+                    TaskCustomIntField.CUSTOM_INT_7,
+                    new IntInterval[] {new IntInterval(null, null), new IntInterval(null, null)}),
+                Pair.of(
+                    TaskCustomIntField.CUSTOM_INT_8,
+                    new IntInterval[] {new IntInterval(123, 122)}));
```

Comment:
split "Both intervals invalid" and "single interval invalid"

Reply:
Ok :)